cpp/src/parquet/encoding_test.cc

```diff
@@ -1383,7 +1383,7 @@ class TestByteStreamSplitEncoding : public TestEncodingBase<Type> {
     encoder->PutSpaced(draws_, num_values_, valid_bits, valid_bits_offset);
     encode_buffer_ = encoder->FlushValues();
     ASSERT_EQ(encode_buffer_->size(), physical_byte_width() * (num_values_ - null_count));
-    decoder->SetData(num_values_ - null_count, encode_buffer_->data(),
+    decoder->SetData(num_values_, encode_buffer_->data(),
```

Comment:
Ouch. Can we also improve the docstring for `Decoder::SetData`, to make it clear that the `num_values` parameter also includes null values?

Reply:
done. This part is so tricky, I may revisit it later